src/blk/kernel/KernelDevice.cc

```diff
@@ -85,13 +85,29 @@ KernelDevice::KernelDevice(CephContext* cct, aio_callback_t cb, void *cbpriv, ai
   }
 }
 
+int KernelDevice::choose_fd(blk_access_mode_t buffered, int write_hint) const
+{
+  assert(write_hint >= WRITE_LIFE_NOT_SET && write_hint < WRITE_LIFE_MAX);
+#if defined(F_SET_FILE_RW_HINT)
+  if (!enable_wrt)
+    write_hint = WRITE_LIFE_NOT_SET;
+#else
+  // Without WRITE_LIFE capabilities, only one file is used.
+  // And rocksdb sets this value also to > 0, so we need to catch this here
+  // instead of trusting rocksdb to set write_hint.
+  write_hint = WRITE_LIFE_NOT_SET;
+#endif
+  return buffered == blk_access_mode_t::BUFFERED ? fd_buffereds[write_hint] 
+				      : fd_directs[write_hint];
+}
+
 int KernelDevice::_lock()
 {
-  dout(10) << __func__ << " " << fd_directs[WRITE_LIFE_NOT_SET] << dendl;
+  dout(10) << __func__ << " " << choose_fd(blk_access_mode_t::DIRECT) << dendl;
```

Comment:
please move this line down after `int fd = choose_fd(blk_access_mode_t::DIRECT);` and use `fd` instead.